### PR TITLE
useSWR refactor

### PR DESCRIPTION
**Pull request by sahedw**
Switch all normal fetches with useSWR for better implementation and UX.
Write custom useSWR hook.
Replace every fetch on global and component level with custom hook (if possible).

### CODE REVIEW

File: hooks/useSWRFetch.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import useSWR from "swr";
+
```
Comment (actyvystom) [inline hooks/useSWRFetch.js:2]: Looks like you created your hook just to have your `fetcher` function available. Do you know about [SWRConfig](https://swr.vercel.app/docs/global-configuration)? Using this module, you can get rid of your own hook and use the `useSWR` hook directly

**Review by actyvystom — COMMENTED** [review]
Nice work! 
Please see my suggestion regarding your custom hook, might be more flexible to use.

File: hooks/useSWRFetch.js (hunk 1 above)
Comment (sahedw) [reply hooks/useSWRFetch.js:2]: Oh you're right! Complete forgot about that. I'll fix it immediately. Thanks.

**Review by JanaDrieschner — APPROVED** [review]
rechecked the issues

**Pull request merged**